sormas-backend/src/main/java/de/symeda/sormas/backend/caze/CaseService.java

```diff
@@ -1201,6 +1202,16 @@ public Predicate createUserFilter(CaseQueryContext caseQueryContext, CaseUserFil
 
 		filter = CriteriaBuilderHelper.or(cb, filter, filterResponsible);
 
+		if (featureConfigurationFacade.isPropertyValueTrue(FeatureType.LIMITED_SYNCHRONIZATION, FeatureTypeProperty.EXCLUDE_NO_CASE_CLASSIFIED_CASES)
+			&& RequestContextHolder.isMobileSync()) {
+			final Predicate limitedCaseSyncPredicate = CriteriaBuilderHelper.and(
```

Comment:
I can't imagine that this works as expected. The idea is to exclude cases with the criteria that is specified in the issue description from the list of cases that are returned, but this code doesn't look like it would do that, but instead that it WOULD only synchronise cases that are not classified as not a case, haven't been reported by the user who does the synchronisation, but at the same time have been created in the mobile app. However, since this is then added to the filter with an `or`, I believe that it won't actually have any effect, would it?

I believe that this code needs to be replaced with something like the following:

```suggestion
final Predicate limitedCaseSyncPredicate = cb.not(cb.and(
  cb.equal(casePath.get(Case.CASE_CLASSIFICATION), CaseClassification.NO_CASE),
  cb.or(
    cb.notEqual(casePath.get(Case.REPORTING_USER), currentUser),
    cb.isNull(casePath.get(Case.CREATION_VERSION)))));
filter = CriteriaBuilderHelper.and(cb, filter, limitedCaseSyncPredicate);
```